bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/scanner/ResumingStreamingResultScanner.java

```diff
@@ -87,25 +103,82 @@ public FlatRow next() throws IOException {
       try {
         FlatRow result = currentDelegate.next();
         if (result != null) {
-          retryHandler.update(result);
+          requestManager.updateLastFoundKey(result.getRowKey());
+          // We've had at least one successful RPC, reset the backoff and retry counter
+          currentErrorBackoff = null;
+          timeoutRetryCount = null;
```

Comment:
There's a non trivial (relatively speaking) cost to building the Backoff object, which IIRC, promoted this us to set the AtomicInteger and Backoff object at failure time rather than incur the creation cost for the vast majority of successful operations.

I just tested AtomicInteger creation, but the cost is less than 1 nanosecond per create.  a new Backoff object is 75 nanoseconds. 

I think you're right about thread safety being a potential concern.  Do you think synchronized handleScanTimeout and handleIOException is enough?